### DISCUSSION TITLE
`withEscapedDependencies` doesn't work sometimes.

### DISCUSSION THREAD

**Discussion by dankinsoid** [pointfreeco/swift-dependencies#166] [General]
### Description

I encountered a strange bug. 
I've created a "Command" wrapper for closures and added an extension to capture dependencies with `withEscapedDependencies`. This extension only worked after I moved it to the same target where it's used.
To reproduce the issue, I first attempted in a clean SPM package but failed. Then, I set up a tuist project similar to my app, where the bug reappeared, but only in the test target, whereas in my original project, it manifested in the app itself.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Expected behavior

`withEscapedDependencies` works properly everywhere.

### Actual behavior

Dependencies are lost.

### Steps to reproduce

1. Download [my example project](https://github.com/dankinsoid/DepsBug).
2. [Install](https://docs.tuist.io/tutorial/get-started/) tuis.
3. go to the project root
4. run `tuist fetch`
5. run `tuist generate`
6. run `test_escaped_dependencies` tests. Bug is unstable, sometimes tests passes, cleaning build folder helps.

### Dependencies version information

1.1.5

### Destination operating system

iOS 17

### Xcode version information

15.0.1

### Swift Compiler version information

```shell
5.9
```

**Comment by dankinsoid** [comment]
There are two debug screens, I used a breakpoint to figure out what happened. 
The `areDependenciesLost` property was `false` at one frame and became `true` when I select the next one.
<img width="917" alt="Снимок экрана 2023-12-21 в 12 03 32" src="https://github.com/pointfreeco/swift-dependencies/assets/30962149/294e7768-1fd9-4ea5-9b36-f55e7db43213">
<img width="973" alt="Снимок экрана 2023-12-21 в 12 03 50" src="https://github.com/pointfreeco/swift-dependencies/assets/30962149/9cacedca-4ec6-477d-a59c-b3c38d04a6e3">

**Comment by dankinsoid** [comment]
When I move my `withEscapedDependencies` extension from `Common` to `TransferMusic` target everything works fine.

**Comment by mbrandonw** [comment]
Hi @dankinsoid, can you please provide a reproduction of the problem without using Tuist in order to rule out a problem with that tool? We’ve had reports in the past of problems that turned out to be a subtlety of using Tuist.

**Comment by dankinsoid** [comment]
@mbrandonw I tried to reproduce the issue without tuist twice and failed.

**Comment by mbrandonw** [comment]
Hi @dankinsoid, in that case I am going to convert this to a discussion. If I have time later to take a look I will, but maybe someone more familiar with Tuist can look into it too.

**Comment by mbrandonw** [comment]
There have been other discussions about Tuist that you may find helpful: https://github.com/pointfreeco/swift-dependencies/issues/75#issuecomment-1543317888
